### DISCUSSION TITLE
SoaRecord timed out or is a transient error

### DISCUSSION THREAD

**Discussion by tore90turris** [MichaCo/DnsClient.NET#164] [Q&A]
Hi MichaCo,
In the meantime, congratulations on the work you are doing.
Currently, I'm using the DnsClient 1.6.1 version (in .Net Framework 4.8), I wanted to report an error that gives very often when I try to read the SOA RECORD in order to get the Serial Number

```
var zone = "AAA.test.";
var endpoint = new IPEndPoint(IPAddress.Parse(IPServer), 53);
var dnsServer =  new LookupClient(endpoint);
var result = dnsServer.Query(zone, QueryType.SOA).Answers.FirstOrDefault() as SoaRecord;
```
Very often I get the error:
`Query 39341 => AAA.test. IN SOA on XX.XX.XX.XX:53 timed out or is a transient error.`

The strange thing to me is the following: If I try to read the type A records first, I am much more likely to succeed in reading the SOA record if I can do it immediately afterwards.
Could it be a coincidence?

Could you help me somehow?

**Comment by MichaConrad** [comment]
I cannot think of anything which would be related to the record type, which would cause a timeout.
Timeouts almost always an issue with the DNS Server / connection. 
So, I'm not sure how I can help you with that because that's not in my control ^^

If you have access to logs of the DNS Server, you should probably look into that.
General network problems with the internet connection or anything else in your network might also cause random timeouts, who knows, its really hard to tell, at least there is nothing I or the library can do to figure that out.

**Reply by magicjude** [reply]
How to make the Query method not throw exceptions under any circumstances? I don't want to use try catch.

**Reply by magicjude** [reply]
I am using version 1.7,. netframework 4.5. Thank you

**Reply by MichaConrad** [reply]
> How to make the Query method not throw exceptions under any circumstances? I don't want to use try catch.

@magicjude  that's not going to happen. The library has to do network calls and those might fail, with a timeout or connection problem or what not. Those errors will never be handled and hidden by the library and you have to deal with that.
